pettingzoo/mpe/scenarios/group_spread.py

```diff
@@ -48,44 +47,37 @@ def make_world(self, groups, num_colors=None, reward_per_group=True, randomise_a
 
     def reset_world(self, world, np_random):
         if self.randomise_all_colors:
-            # generate colors if they are randomised for each episode
-            self.colors = [np.random.random(3) for _ in self.groups]
-            color_idxs = np.arange(len(self.groups))
-        else:
-            # assign one of the available colors to a group
-            color_idxs = np.random.choice(self.num_colors, len(self.groups), replace=False)
-
-        # random properties for agents
+            # TODO: do nicer to avoid overlap here?
+            # randomly generate new colors for each episode
+            self.colors = [np.random.random(3) for _ in range(self.num_colors)]
```

Comment:
Not sure I see the point here? There should be one colour for each group. The num_colors parameter is used to specify how many colours are available to chose from, which in the case of randomise_all_colors should be all possible colours. I think this is only required if we change the colour generator to use fixed colours/interval as suggested above, otherwise I think it works. Agree that maybe the parameters need new names as we've added new functionality it may have got confusing.